frontend/src/pages/AccountForm.tsx

```diff
@@ -1,134 +1,196 @@
-import React, { useState } from 'react';
-import { makeStyles } from '@material-ui/core/styles';
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+// import { makeStyles } from '@material-ui/core/styles';
 import Button from '@material-ui/core/Button';
 import List from '@material-ui/core/List';
 import ListItem from '@material-ui/core/ListItem';
 import Typography from '@material-ui/core/Typography';
-import { blue } from '@material-ui/core/colors';
+// import { blue } from '@material-ui/core/colors';
 import EditIcon from '@material-ui/icons/Edit';
 import TextField from '@material-ui/core/TextField';
+import axios from 'axios';
 
-const emails = ['username@gmail.com', 'user02@gmail.com'];
+// const emails = ['username@gmail.com', 'user02@gmail.com'];
 
 const initialState = {
-    name: "Sam Ip",
-    email: "samip@email.com",
-    Major: "Computer Science",
-    courses: [
-        'CPSC 110', 'CPSC 121', 'CPSC 210', 'CPSC 213', 'CPSC 221', 'CPSC 310'
-    ]
-}
+  name: 'Sam Ip',
+  email: 'samip@email.com',
+  major: 'Computer Science',
+  courses: [
+    'CPSC 110', 'CPSC 121', 'CPSC 210', 'CPSC 213', 'CPSC 221', 'CPSC 310',
+  ],
+};
 
-export default function AccountForm() {
-    const [open, setOpen] = React.useState(false);
-    const [profile, setProfile] = React.useState(initialState);
-    const [editName, toggleEdit] = React.useState(false);
-    const [editEmail, toggleEmail] = React.useState(false);
-    const [editMajor, toggleMajor] = React.useState(false);
-    const [inputs, setInputs] = React.useState({});
-    const [profileName, setProfileName] = React.useState(initialState.name);
-    const [profileEmail, setProfileEmail] = React.useState(initialState.email);
-    const [profileMajor, setProfileMajor] = React.useState(initialState.Major);
```

Comment:
file to be refactored later, to unify state, use redux instead of local state, and extract api calls